build.js

```diff
@@ -422,655 +699,812 @@ function buildSupportProjects(api, config) {
+                "app-debug.apk"
+            ];
+
+            var apkDirDebug = path.join(outputPath, "../../"+shortName+"/app/build/outputs/apk/debug/");
+            return spawnWithLogger(api, process.env.ANDROID_HOME + '/build-tools/27.0.3/apksigner', signArgsRelease, {cwd: apkDirDebug})
+        }
+        else {  // sign debug apk with default Android debug keys
+            var keyPath = path.join(process.env['HOME'], '.android', 'debug.keystore');
+            signArgsDebug = [
+                "-sigalg", "MD5withRSA", "-digestalg", "SHA1",
+                "-keystore", keyPath, "-storepass", "android",
+                "-signedjar", "app-debug-aligned.apk",
+                "app-signed.apk", "androiddebugkey"
+            ];
+
+            alignArgsDebug = [
+                "-f", "-v", "4", "app-debug.apk", "app-debug-aligned.apk"
+            ];
+
+            var apkDirDebug = path.join(outputPath, "../../" + shortName + "/app/build/outputs/apk/debug/");
+            return spawnWithLogger(api, process.env.ANDROID_HOME + '/build-tools/27.0.3/zipalign', alignArgsDebug, {cwd: apkDirDebug})
+                .then(function () {
+                    spawnWithLogger(api, process.env.ANDROID_HOME + '/build-tools/27.0.3/apksigner', signArgsDebug, {cwd: apkDirDebug})
+                })
+        }
+
+    } else {
+        var keystore = process.env['DEVKIT_ANDROID_KEYSTORE'];
+        if (!keystore) { throw new BuildError('missing environment variable DEVKIT_ANDROID_KEYSTORE'); }
+
+        var storepass = process.env['DEVKIT_ANDROID_STOREPASS'];
+        if (!storepass) { throw new BuildError('missing environment variable DEVKIT_ANDROID_STOREPASS'); }
+
+        var keypass = process.env['DEVKIT_ANDROID_KEYPASS'];
+        if (!keypass) { throw new BuildError('missing environment variable DEVKIT_ANDROID_KEYPASS'); }
+
+        var key = process.env['DEVKIT_ANDROID_KEY'];
+        if (!key) { throw new BuildError('missing environment variable DEVKIT_ANDROID_KEY'); }
+
+        signArgs = [
+            "sign", "--ks", keystore, "--ks-pass", "pass:"+storepass, "--key-pass", "pass:"+keypass,
+            "--ks-key-alias", key, "--v1-signing-enabled", "true", "--v2-signing-enabled", "false", "--verbose",
+            "app-release-aligned.apk" //shortName + "app-release-unsigned.apk", key
+        ];
+
+        alignArgs = [
+            "-f", "-v", "4", "app-release-unsigned.apk", "app-release-aligned.apk" //shortName + "-unaligned.apk", shortName + "-aligned.apk"
+        ];
+
+        var scheme = (config.debug ? "debug" : "release");
+        var apkDir = path.join(outputPath, "../../"+shortName+"/app/build/outputs/apk/"+scheme+"/");
+        return spawnWithLogger(api, process.env.ANDROID_HOME +'/build-tools/27.0.3/zipalign', alignArgs , {cwd: apkDir})
+            .then(function () {
+                spawnWithLogger(api, process.env.ANDROID_HOME +'/build-tools/27.0.3/apksigner', signArgs, {cwd: apkDir})
```

Comment:
we are hardcoding the path(`build-tools/27.0.3/apksigner`)
Can that be determined automatically ?

Reply:
This shel pass well because we use 27.0.3 in build.gradle. Otherwise it will be reasonable to synchronize build.gradle and build.js setting